common/core/desktop/meson.build

```diff
@@ -23,26 +23,33 @@ endif
 py = import('python3')
 python = py.find_python()
 
+message('host_machine.system(): ' + host_machine.system())
+message('compiler.get_id(): ' + compiler.get_id())
 # Rust library path search
 
-if host_machine.system() == 'windows'
-  if host_machine.cpu_family() == 'x86'
-    rust_platform = 'x86' #i686-pc-windows-msvc in rust
-  else
-    rust_platform = 'x64' #x86_64-pc-windows-msvc in rust
-  endif
+if host_machine.system() == 'emscripten'
+  rust_platform = 'wasm'
 else
-  if host_machine.system() == 'darwin'
-    # todo we need to support M1 targets as well, eventually
-    rust_platform = 'arch'
-  else #linux
-    # we only build for our current arch
-    rust_platform = 'arch'
+  if host_machine.system() == 'windows'
+    if host_machine.cpu_family() == 'x86'
+      rust_platform = 'x86' #i686-pc-windows-msvc in rust
+    else
+      rust_platform = 'x64' #x86_64-pc-windows-msvc in rust
+    endif
+  else
+    if host_machine.system() == 'darwin'
+      # todo we need to support M1 targets as well, eventually
+      rust_platform = 'arch'
+    else #linux
+      # we only build for our current arch
+      rust_platform = 'arch'
+    endif
   endif
 endif
```

Comment:
I don't know this structure, does it do something reasonable if none of these match?

Reply:
Yes, we are doing special casing for Windows and macOS, but everything else will use the default architecture for the platform (hence using the name `arch`).